File: testdata/lint/py/falsenegatives.py
# falsenegatives.py documents the capability lint's known, accepted limits
# (spec 1.3: lint is heuristic and advisory, not proof of absence). Two
# known false negatives live here: a dynamically imported module named by a
# variable is never followed, and a capability hidden behind eval is never
# scanned, because DetectPython matches literal text line by line and never
# evaluates or interprets the source it reads. One known false positive
# lives here too: a commented out import still matches the same line
# anchored pattern a live import would, because DetectPython does not parse
# comments out; that is an accepted, documented limitation, not a defect,
# the same posture DetectJS documents for a commented out require.

import importlib

CAPABILITY_MODULE = "subprocess"


def load_capability():
    return importlib.import_module(CAPABILITY_MODULE)


def load_hidden(encoded):
    return eval(encoded)


# import requests

File: testdata/lint/js/falsenegatives.ts
// falsenegatives.ts documents the capability lint's known, accepted limits
// (spec 1.3: lint is heuristic and advisory, not proof of absence). Two
// known false negatives live here: a dynamic import whose specifier is a
// variable is never followed, and a capability hidden behind eval is never
// scanned, because DetectJS matches literal text line by line and never
// evaluates or interprets the source it reads. One known false positive
// lives here too: a commented out call still matches the same line
// anchored pattern a live call would, because DetectJS does not parse
// comments out; that is an accepted, documented limitation, not a defect.

async function loadDynamic(moduleName) {
	return import(moduleName);
}

function loadHidden(hexDecoded) {
	return eval(hexDecoded);
}

// require("http")
